### PR TITLE
Appdata

### PR DESCRIPTION
**Pull request by har9862**
+ Migrated to the latest AppStream specs
+ Updated project_license to a valid [SPDX](https://spdx.org/licenses/) license identifier
+ Added category
+ Added url for bugtracker, faq, help, translate and contact

### CODE REVIEW

File: org.filezillaproject.Filezilla.appdata.xml [hunk 1]
```diff
@@ -1,11 +1,11 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <!-- Copyright 2014 Ryan Lerch <rlerch@redhat.com> -->
 <!-- Copyright 2016 Tim Kosse <tim.kosse@filezilla-project.org> -->
-<application>
-  <id type="desktop">org.filezillaproject.Filezilla.desktop</id>
+<component type="desktop">
```
Comment (bilelmoussaoui) [inline org.filezillaproject.Filezilla.appdata.xml]:
```suggestion
<component type="desktop-application">
```

File: org.filezillaproject.Filezilla.appdata.xml [hunk 2]
```diff
@@ -30,6 +30,9 @@
       <li>Remote file search</li>
     </ul>
   </description>
+  <categories>
```
Comment (bilelmoussaoui) [inline org.filezillaproject.Filezilla.appdata.xml]: Not needed, they are grabbed from the desktop file using appstream-compose.

File: org.filezillaproject.Filezilla.appdata.xml (hunk 1 above)
Comment (TingPing) [reply org.filezillaproject.Filezilla.appdata.xml]: Doesn't really matter in practice.

File: org.filezillaproject.Filezilla.appdata.xml [hunk 3]
```diff
@@ -1,11 +1,11 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <!-- Copyright 2014 Ryan Lerch <rlerch@redhat.com> -->
 <!-- Copyright 2016 Tim Kosse <tim.kosse@filezilla-project.org> -->
-<application>
-  <id type="desktop">org.filezillaproject.Filezilla.desktop</id>
+<component type="desktop">
+  <id>org.filezillaproject.Filezilla.desktop</id>
```
Comment (TingPing) [inline org.filezillaproject.Filezilla.appdata.xml]:
```suggestion
  <id>org.filezillaproject.Filezilla</id>
```

And add `launchable`

**Comment by TingPing** [comment]
Try to keep this in sync with upstream too.

**Comment by nedrichards** [comment]
From the error:
> org.filezillaproject.Filezilla.desktop: AppData problem: tag-invalid : <url> type invalid [unknown]
Error loading AppData file: AppData file /app/share/appdata/org.filezillaproject.Filezilla.appdata.xml was not valid
Error: ERROR: appstream-compose failed: Child process exited with code 1
FB: unmounting rofiles-fuse /srv/buildbot/worker/build-x86_64-2/build/.flatpak-builder/rofiles/rofiles-IZtxTo

File: org.filezillaproject.Filezilla.appdata.xml (hunk 3 above)
Comment (nedrichards) [reply org.filezillaproject.Filezilla.appdata.xml]: This is in the provides section currently. https://github.com/flathub/org.filezillaproject.Filezilla/pull/28/files#diff-cbb9610615681de4b45d930971da7381R94 shouldn't be in both certainly.

Comment (TingPing) [reply org.filezillaproject.Filezilla.appdata.xml]: No `org.filezilla-project.FileZilla` != `org.filezillaproject.Filezilla`. Remember upstream named it different than here so both are just screwed.

**Comment by har9862** [comment]
Thanks for the review everyone, also spotted a mistake I made for the project license. Patch updated.

> From the error:
> 
> > org.filezillaproject.Filezilla.desktop: AppData problem: tag-invalid :  type invalid [unknown]

I think that's caused by line 89:
`<url type="contact">https://filezilla-project.org/contact.php</url>`

I can remove this if needed for passing build, but it should be valid according to the specs [here](https://www.freedesktop.org/software/appstream/docs/chap-Metadata.html#tag-url).

**Pull request merged**